### DISCUSSION TITLE
Make different random setups saveable under favorites, and include folders

### DISCUSSION THREAD

**Discussion by viertelb** [LMS-Community/slimserver#654] [General]
At the moment (8.1.1) it is not possible to save more than one distinct "random titles" setup. It would be good settings for the genre could be saved (and named) and then put into the favorites. 

The same might go for "random albums".

Also, it is not possible to random play a certain folder, which would also be nice to have.

**Comment by michaelherger** [comment]
`randomplay://tracks?genres=rock,pop,jazz` should allow you to favorite genre mixes. Replace `tracks` with `albums` for an album mix.

What do you mean by "random play a certain folder"? Pick a random folder and play its content? Pick a folder and play its content randomly?

**Comment by viertelb** [comment]
Chosing a specific folder from MyMusic, and let LMS play its content randomly.

How to apply your code snippet? Is it an extension to my LMS URL? My browser shows `192.168.10.40:9000`.

**Comment by michaelherger** [comment]
I'm still not sure what "a specific folder from MyMusic" is. Are you referring to a category like Artists, or Albums? Or a folder from Browse Musicfolders?

You can use the URL I posted to create a favorite, or save it as a preset using the Preset Editor plugin.

**Comment by viertelb** [comment]
Yes, Browse Musicfolders. Not a category, but the folders on the file system. 

How to apply the URL? I know this is wrong: `192.158.10.40:9000/randomplay://tracks?genres=rock,pop,jazz`

**Comment by michaelherger** [comment]
Just `randomplay://tracks?genres=rock,pop,jazz`, no http or IP etc.

**Comment by viertelb** [comment]
How would my browser know to which IP to apply this? My LMS is running on a different server in my LAN. Putting in just that line will pop up a google answer. :)

**Comment by mherger** [comment]
You don't enter that URI in your browser, but in LMS' "Tune in to..." field, or the favorites etc. It only works in LMS.

**Comment by viertelb** [comment]
Is this "URL laden..." in german? 

If I enter it there I can then add this to the playlist or play it right away. But I don't see an option to save it as a favorite. I run 8.1.1.

**Comment by michaelherger** [comment]
Yes, that's it.

You can edit favorites by clicking the small pencil on the right hand side of the menu item in the default LMS web UI.

**Comment by viertelb** [comment]
Ok, so you mean I can `edit` the favorites manually in the web UI and add in `randomplay://tracks?genres=rock,pop,jazz` in the `URL` field which then opens up on the right side (left is `name`). 

If I do this I can add several new favorites items (with different variables, e.g. `genres=rock`, `genres=rock,pop`), yes. 

But when playing them those settings seem not to be used. I save a favorite with just `rock`, this will be shown as `randomplay://tracks?genres=rock` when chosen, but it will generate 5 items in the (cleared) playlist from all genres.

**Comment by michaelherger** [comment]
Please note that the genre match is exact and case sensitive. It might be `Rock` instead... And it must exist in your collection, of course.

**Comment by viertelb** [comment]
Yes, that's it. It is working well now. I suggest adding a hint somewhere that favorites can be added by clicking on the small pencil on the right hand side of the menu item, and a template with this URI (and others that would work).
